developer/src/kmc-copy/src/KeymanProjectCopier.ts

```diff
@@ -0,0 +1,561 @@
+
+    const lines = new TextDecoder().decode(result.artifacts[filename].data).replaceAll(/\r\n/g, '\n').replaceAll(/\r/g, '\n').split('\n');
+    for(let i = 0; i < lines.length; i++) {
+      const line = lines[i];
+      for(const r of storeRegexes) {
+        const m = line.match(r.re);
+        if(m) {
+          const newRelativePath = this.copySubFileAndGetRelativePath(project, filename, m[2], outputPath, source, result);
+          lines[i] = line.replace(replacementRegex, `$1${newRelativePath}$4`);
+        }
+      }
+    }
+
+    // rewrite fixups inside the source file
+    result.artifacts[filename].data = new TextEncoder().encode(lines.join('\n'));
+
+    return true;
+  }
+
+  private copyKpsSourceFile(project: KeymanDeveloperProject, filename: string, outputPath: string, source: string, result: CopierResult): boolean {
+    if(!this.copySourceFile(project, filename, outputPath, source, result)) {
+      return false;
+    }
+
+    const reader = new KpsFileReader(this.callbacks);
+    const kps = reader.read(result.artifacts[filename].data);
+    if(!kps) {
+      this.callbacks.reportMessage(CopierMessages.Error_PackageFileCouldNotBeRead({filename}));
+      return false;
+    }
+
+    if(kps.Package?.Files?.File.length) {
+      for(const file of kps.Package.Files.File) {
+        // const subFilename = this.callbacks.resolveFilename(filename, file.Name);
+        const newRelativePath = this.copySubFileAndGetRelativePath(project, filename, file.Name, outputPath, source, result);
+
+        const remap = (ref?: string) => ref == file.Name ? newRelativePath : ref;
+
+        // Update all other file links in the .kps
+        kps.Package.Options.GraphicFile = remap(kps.Package.Options.GraphicFile);
+        kps.Package.Options.LicenseFile = remap(kps.Package.Options.LicenseFile);
+        kps.Package.Options.MSIFileName = remap(kps.Package.Options.MSIFileName);
+        kps.Package.Options.ReadMeFile = remap(kps.Package.Options.ReadMeFile);
+        kps.Package.Options.WelcomeFile = remap(kps.Package.Options.WelcomeFile);
+        if(kps.Package?.Keyboards?.Keyboard?.length) {
+          for(const keyboard of kps.Package.Keyboards?.Keyboard) {
+            keyboard.DisplayFont = remap(keyboard.DisplayFont);
+            keyboard.OSKFont = remap(keyboard.OSKFont);
+            if(keyboard.WebDisplayFonts?.Font?.length) {
+              for(const font of keyboard.WebDisplayFonts.Font) {
+                font.$.Filename = remap(font.$.Filename);
+              }
+            }
+            if(keyboard.WebOSKFonts?.Font?.length) {
+              for(const font of keyboard.WebOSKFonts.Font) {
+                font.$.Filename = remap(font.$.Filename);
+              }
+            }
+
+            // Rewrite keyboard id
+            if(keyboard.ID == this.sourceId) {
+              keyboard.ID = this.outputId;
+            }
+          }
+        }
+
+        if(kps.Package?.LexicalModels?.LexicalModel?.length) {
```

Comment:
Are we still flagging an error if a kps includes both keyboard and lexical model?

Reply:
Only in the compiler, not in the copying tool. If there are problems in the source project, they'll be replicated in the target project.